### PR TITLE
Creating Responsive Line and Bar charts with React and Semantic-ui

### LINKED ISSUE #4724: Creating Responsive Line and Bar charts with React and Semantic-ui
### Introduction:
Graphic illustration is an essential form of content presentation in front-end web development. Line and bar charts may have various applications such as, to show statistics of user engagements, user growth, ratings, etc. React.js, semantic-ui and CSS can be used to generate various line and bar charts as we will discuss in this article.
### Key Takeaways:
It is expected that at the completion of this article the reader will be equipped with the knowledge and necessary skill to understand and perform and the following: 
i.	Installing and adding the semantic-ui dependency to your web project
ii.	Getting started with React functional components
iii.	Creating reusable components with React.js
iv.	Generating Line and Bar charts with React.js
v.	Styling the components with CSS and semantic-ui
### References:
https://react.semantic-ui.com/elements/

#### Article quality
This article will walk the reader through the various steps required to generate line and bar charts from an array of numbers used as coordinates. The article will also teach the reader how to create a reusable chart component. Using semantic UI design libraries in creating charts is a unique concept that will be discussed in-depth.

### PR DESCRIPTION
**Pull request by FredBenson**
#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

this closes #4724

### CODE REVIEW

**Comment by ahmadmardeni1** [comment]
Plagiarism check done ✅

**Comment by ahmadmardeni1** [comment]
Why did you delete our template before submitting your PR? @FredBenson

**Comment by FredBenson** [comment]
@ahmadmardeni1 
I apologize, but it wasn't there when I submitted. Please could you help me fix it?

**Comment by ahmadmardeni1** [comment]
I edited your initial comment and added the template. Check what matches your article please. @FredBenson

**Review by jerimkaura — COMMENTED** [review]
Hello @FredBenson :wave: 

Thank you for your great article. Take note of the changes I made to polish your future articles.

Proceeding to tech review :+1:

**Review by jerimkaura — COMMENTED** [review]
@FredBenson  Kindly provide an author image in the same folder as your profile.

**Comment by FredBenson** [comment]
hello 
@jerimkaura I have added the author image as requested, for some reason it disappeared.

File: content/articles/creating-responsive-line-and-bar-charts-with-semantic-ui-and-react/index.md [hunk 1]
```diff
@@ -0,0 +1,286 @@
+
+Creating Responsive Line and Bar charts with React and Semantic-ui
+
+### Introduction
+A graphic illustration is a basic form of content presentation in front-end web development. For example, line and bar charts may have various applications to show statistics of user engagements, user growth, ratings, etc. Creating the perfect responsive charts for your company or personal website will be made simple and easy to implement.
```
Comment (jerimkaura) [inline content/articles/creating-responsive-line-and-bar-charts-with-semantic-ui-and-react/index.md:5]: Explain what will make it simple to implement

File: content/articles/creating-responsive-line-and-bar-charts-with-semantic-ui-and-react/index.md [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+### Key Takeaways
+It is expected that after this article, the reader will be equipped with the knowledge and necessary skill to understand and perform and the following:
+- Getting started with React functional components
+- Installing and adding the semantic-UI dependency to your web project
+- Creating reusable components with React.js
+- Generating Line and Bar charts with react-chartjs-2
+- Styling the components with CSS and semantic-ui
+
+### Prerequisite
+This tutorial requires intermediate knowledge of React.js, CSS, and any modern web styling libraries similar to Semantic-UI. So, for absolute beginners, take a moment and enroll in a crash course.
```
Comment (jerimkaura) [inline content/articles/creating-responsive-line-and-bar-charts-with-semantic-ui-and-react/index.md:20]: Provide a link for the course or tutorial.

File: content/articles/creating-responsive-line-and-bar-charts-with-semantic-ui-and-react/index.md [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+
+### Getting started with React functional components
+React functional component is a recent enhancement to React.js, the former being a class-based component. A functional component is a react function that accepts a `props` and returns a JSX (JavaScript syntax extension). For this tutorial, we will adopt react functional components for our application.
```
Comment (jerimkaura) [inline content/articles/creating-responsive-line-and-bar-charts-with-semantic-ui-and-react/index.md:23]: Tell the user what a prop is.

**Review by jerimkaura — CHANGES_REQUESTED** [review]

**Comment by FredBenson** [comment]
@jerimkaura done as requested

**Review by jerimkaura — APPROVED** [review]
Hello @FredBenson  

Thank you for being patient up to this time :100: Your article was clear and precise. Take note of the change any reviewer or I make or will make to polish your future article even more.

This article is ready for final review. :rocket:

**Review by lalith1403 — CHANGES_REQUESTED** [review]
I have highlighted a few places of errors that we expect. Sentences should be short and crisp. Variables highlighted properly. Terminologies used correctly and explained if required. 

Most importantly, fact-checking and making sure any statements are backed by facts.

Please make the above changes and submit a new PR. I will be closing this PR for now to make room for other PRs. Thank you.

cc: @jerimkaura Please make note of the changes made. All articles should be factually, grammatically and formatted correctly. Thank you.

**Pull request closed without merging**